AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.java

```diff
@@ -597,6 +597,11 @@ protected void onCreate(Bundle savedInstanceState) {
         setContentView(R.layout.card_browser);
         initNavigationDrawer(findViewById(android.R.id.content));
         startLoadingCollection();
+
+        if (getIntent().getBooleanExtra("called_from_js_api", false)) {
+            mSearchTerms = getIntent().getStringExtra("search_query");
+            searchCards();
+        }
```

Comment:
If you just checked for the search_query Intent extra, and used it if non-null, then all the "called_from_js_api" stuff can disappear, making this a much smaller / easier to understand change

```suggestion

        if (getIntent().getStringExtra("search_query") != null) {
            mSearchTerms = getIntent().getStringExtra("search_query");
            searchCards();
        }
```